numpyro/infer/elbo.py

```diff
@@ -311,53 +312,166 @@ class RenyiELBO(ELBO):
         Here :math:`\alpha \neq 1`. Default is 0.
     :param num_particles: The number of particles/samples
         used to form the objective (gradient) estimator. Default is 2.
+    :param callable create_plates: An optional function inputing the same
+        ``*args,**kwargs`` as ``model()`` and returning a :class:`numpyro.plate`
+        or iterable of plates. Plates not returned will be created
+        automatically as usual. This is useful for data subsampling: the indices
+        of those plates will be the same across particles.
+
+    Example::
+
+        def model(data):
+            with numpyro.plate("batch", 10000, subsample_size=100):
+                latent = numpyro.sample("latent", dist.Normal(0, 1))
+                batch = numpyro.subsample(data, event_dim=0)
+                numpyro.sample("data", dist.Bernoulli(logits=latent), obs=batch)
+
+        def guide(data):
+            w_loc = numpyro.param("w_loc", 1.)
+            w_scale = numpyro.param("w_scale", 1.)
+            with numpyro.plate("batch", 10000, subsample_size=100):
+                batch = numpyro.subsample(data, event_dim=0)
+                loc = w_loc * batch
+                scale = jnp.exp(w_scale * batch)
+                numpyro.sample("latent", dist.Normal(loc, scale))
+
+        def create_plates(data):
```

Comment:
i guess the user will wonder why this information can't be extracted from the model/guide...

Reply:
I think sometimes, users will want to compute MC approximation before computing importance weights. If that usage case is rare, we can assume that subsample indices are always the same across particles. In this case, there are seperal ways to assure getting constant seeds for subsample plates across particles. E.g. we can introduce seed_plate handler and apply it before vmapping over particles. WDYT? should we always perform MC after computing the weights?

Reply:
are you referring to a usage case that results in biased elbo estimates?

Reply:
yup. so I just removed it.